File: backend/app_api/domain/entities/DetalleCarrito_entity.py
from dataclasses import dataclass
from app_api.domain.values.timestamp import TimestampModel

@dataclass(kw_only=True)
class DetalleCarritoEntity(TimestampModel):
    id: int | None
    carrito: int
    producto: int
    cantidad: int
    descuento: int | None

File: frontend/src/domain/entities/DetalleCarrito_entity.ts
import { TimeStamp } from "../values/Timestamp";

export class DetalleCarritoEntity extends TimeStamp {
    id: number | null;
    carrito: number;
    producto: number;
    cantidad: number;
    descuento: number | null;

    constructor(id: number | null, carrito: number, producto: number, cantidad: number, descuento: number | null,
        created_at: Date | null = null,
        updated_at: Date | null = null,
        deleted_at: Date | null = null
    ) {
        super(created_at, updated_at, deleted_at);
        this.id = id;
        this.carrito = carrito;
        this.producto = producto;
        this.cantidad = cantidad;
        this.descuento = descuento;
    }
}

